tasks/custom.yml

```diff
@@ -0,0 +1,68 @@
+---
+
+- name: Download custom modules source tar
+  become: yes
+  get_url:
+    url: "{{ item.source }}"
+    dest: "/opt/{{ item.name }}.tar.gz"
+  loop: "{{ nginx_custom_modules }}"
+
+- name: Extract custom modules
+  become: yes
+  unarchive:
+    src: "/opt/{{ item.name }}.tar.gz"
+    dest: /opt
+    remote_src: yes
+  loop: "{{ nginx_custom_modules }}"
+
+- name: Collect current nginx binary version
+  shell: "nginx -v 2>&1 | grep -o '[0-9.]*$'"
```

Comment:
Rather than getting the version number this way, why not just specify this as a variable? That would allow us to pick the version of nginx we want to use, which is part of the flexibility of building from source.

Also I'm not sure how you're able to run `nginx` before it's installed. Does this rely on the `apt` version to be installed as well?

Reply:
I chose to piggyback off of the apt installation which completely sets up nginx to run on Ubuntu 16.04. Building Nginx from source alone doesn't create all the files that are required. That is why i'm getting the nginx version this way, building the source file, then just switching out the binary. I could also use a variable and apt install a specific version if you would like.

Reply:
Also, what is the scenario that we need to install a specific nginx version, and how likely is that scenario?

Reply:
I guess I'm thinking of future confusion that could come if someone notices that nginx is installed via `apt`. I'm not sure how to get around that. I guess we'll just have to let our Ansible roles be the guide.